readme.md

```diff
@@ -16,3 +16,9 @@ Everyone
 ##Test
 
     npm run test
+
+##Optimizations
```

Comment:
One more possible optimization I thought worth noting is to do logical deletes on the energy matrix. This way the cost of deletion goes way down, with the cost of finding the neighbor when recalculating going up some.